Automattic/woocommerce-payments
# Migrate e2e tests for dispute response flows to Playwright

Fixes #9426 

#### Changes proposed in this Pull Request

Migrates the following dispute flow e2e tests from Puppeteer to Playwright

1. Accept a dispute
2. Challenge a dispute with winning evidence
3. Challenge a dispute with losing evidence
4. Save a dispute challenge without submitting evidence
5. View dispute details via disputed order notice
	-  had already been migrated to Playwright (https://github.com/Automattic/woocommerce-payments/blob/007356ab179880100873b412b86051f702c20a32/tests/e2e-pw/specs/merchant/merchant-disputes-view-details-via-order-notice.spec.ts), so this PR removes the respective Puppeteer test

I've also added a couple of helper functions:
- `getAnonymousShopper` to allow stateless shopper browser contexts to allow isolated tests (e.g. separate carts).
- `goToPaymentDetails` for navigation to the payment details screen for a given `paymentIntentId`.

#### Testing instructions





* Ensure Playwright E2E test GH action passes
* Optionally, run these locally
	* `npm run test:e2e-pw disputes-respond` to only run those in this PR

**New tests shown by `npx playwright show-report`**

![image](https://github.com/user-attachments/assets/93e50a24-4a61-4ce3-ae79-a4692a4dc3b1)

-------------------


**Post merge**
